common/metrics/otl.go

```diff
@@ -0,0 +1,77 @@
+package metrics
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+
+	"github.com/ceramicnetwork/go-cas/models"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutmetric"
+	"go.opentelemetry.io/otel/metric"
+	sdk "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+type OTLMetricService struct {
+	meterProvider *sdk.MeterProvider
+	meter         metric.Meter
+}
+
+func NewMetricService(ctx context.Context) (models.MetricService, error) {
+	var resources *resource.Resource
+	// Resources are a special type of attribute that apply to all spans generated by a process.
+	// These should be used to represent underlying metadata about a process that’s non-ephemeral,
+	// for example, the hostname of a process, or its instance ID.
+	resources, err := resource.New(ctx,
+		resource.WithFromEnv(), // pull attributes from OTEL_RESOURCE_ATTRIBUTES and OTEL_SERVICE_NAME environment variables
+		resource.WithAttributes(attribute.String("environment", os.Getenv("ENV"))),
+	)
+	if err != nil {
+		return &OTLMetricService{}, err
+	}
+
+	var exporter sdk.Exporter
+	collectorUrl := os.Getenv("COLLECTOR_URL")
+	if collectorUrl != "" {
+		options := otlpmetrichttp.WithEndpoint(collectorUrl)
+		exporter, err = otlpmetrichttp.New(ctx, options)
+		if err != nil {
+			return &OTLMetricService{}, err
+		}
+	} else {
+		enc := json.NewEncoder(os.Stdout)
+		enc.SetIndent("", "  ")
+		exporter, err = stdoutmetric.New(
+			stdoutmetric.WithEncoder(enc),
+		)
+		if err != nil {
+			return &OTLMetricService{}, err
+		}
+	}
+
+	meterProvider := sdk.NewMeterProvider(
+		sdk.WithResource(resources),
+		sdk.WithReader(sdk.NewPeriodicReader(exporter)),
+	)
+
+	meter := meterProvider.Meter("github.com/ceramicnetwork/go-cas/common/metrics")
```

Comment:
maybe just 'go-cas'

later we can move this to a go observability repo and use caller again